src/MessagePack/MessagePackReader.cs

```diff
@@ -779,8 +779,7 @@ private string ReadStringSlow(int byteLength)
             // We need to concatenate all the bytes together into one array.
             byte[] byteArray = ArrayPool<byte>.Shared.Rent(byteLength);
             ThrowInsufficientBufferUnless(this.reader.TryCopyTo(byteArray));
-            this.reader.Advance(byteArray.Length);
-            string value = StringEncoding.UTF8.GetString(byteArray);
+            string value = StringEncoding.UTF8.GetString(byteArray, 0, byteLength);
```

Comment:
The rented buffer could be much larger than `byteLength`, you want to read a string of size `byteLength`